lua/telescope/sorters.lua

```diff
@@ -38,13 +38,24 @@ Sorter.__index = Sorter
 ---@field highlighter function: Highlights results to display them pretty
 ---@field discard boolean: Whether this is a discardable style sorter or not.
 ---@field score function: Override the score function if desired.
+---@field init function: Function to run when creating sorter
+---@field start function: Function to run on every new prompt
+---@field finish function: Function to run after every new prompt
+---@field destroy function: Functo to run when destroying sorter
 function Sorter:new(opts)
   opts = opts or {}
 
   return setmetatable({
     score = opts.score,
     state = {},
     tags = opts.tags,
+
+    -- State management
+    init    = opts.init,
+    start   = opts.start,
+    finish  = opts.finish,
+    destory = opts.destroy,
```

Comment:
TJ you are driving me crazy right now :sob: typo here: should be `destroy`

Reply:
hahahahahahahaha my bad